pulsar-io/kafka/src/main/java/org/apache/pulsar/io/kafka/KafkaAbstractSource.java

```diff
@@ -64,6 +66,7 @@ public abstract class KafkaAbstractSource<V> extends PushSource<V> {
     private volatile boolean running = false;
     private KafkaSourceConfig kafkaSourceConfig;
     private Thread runnerThread;
+    private final Executor executor = Executors.newSingleThreadExecutor();
```

Comment:
please close this executor. :)

Reply:
changed it to a static variable

Reply:
already remove the executor, and just new one thread if needed